Datastore/src/DatastoreClient.php

```diff
@@ -157,33 +187,57 @@ public function __construct(array $config = [])
     {
         $emulatorHost = getenv('DATASTORE_EMULATOR_HOST');
 
-        $connectionType = $this->getConnectionType($config);
-
         $config += [
             'namespaceId' => null,
             'databaseId' => '',
             'returnInt64AsObject' => false,
             'scopes' => [self::FULL_CONTROL_SCOPE],
             'projectIdRequired' => true,
             'hasEmulator' => (bool) $emulatorHost,
-            'emulatorHost' => $emulatorHost
+            'emulatorHost' => $emulatorHost,
         ];
+        $config = $this->buildClientOptions($config);
+        $config['credentials'] = $this->createCredentialsWrapper(
+            $config['credentials'],
+            $config['credentialsConfig'],
+            $config['universeDomain'],
+        );
+
+        $this->projectId = $this->detectProjectId($config);
 
-        $config = $this->configureAuthentication($config);
-        $this->connection = $connectionType === 'grpc'
-            ? new Grpc($config)
-            : new Rest($config);
+        $this->serializer = new Serializer([], [
+            'google.protobuf.Value' => function ($v) {
+                return $this->flattenValue($v);
+            },
+            'google.protobuf.Timestamp' => function ($v) {
+                return $this->formatTimestampFromApi($v);
+            }
+        ], [], [
+            'google.protobuf.Timestamp' => function ($v) {
+                if (is_string($v)) {
+                    $dt = new \DateTime($v);
+                    return ['seconds' => $dt->format('U')];
+                }
+                return $v;
+            }
+        ]);
+
+        $this->requestHandler = new RequestHandler(
+            $this->serializer,
+            self::GAPIC_KEYS,
```

Comment:
I second @saranshdhingra's opinion - I'd prefer to remove this rather than have it here just to stay consistent with something that doesn't apply. 
```suggestion
            [V1DatastoreClient::class],
```

TBH, I would like to remove `GAPIC_KEYS` for `PubSubClient` as well and just pass these in, as they don't serve any real purpose as a constant (they're not ever reused).